From 841e844f615e17cf18c180963304d14c76b03986 Mon Sep 17 00:00:00 2001
From: Daniel Engberg <daniel.engberg.lists@pyret.net>
Date: Sun, 10 Feb 2019 11:53:02 +0000
Subject: [PATCH] tools/tar: Update to 1.31

Update tar to 1.31
Fixes CVE-2018-20482
Switch to tar.xz tarball
Refresh patches

Signed-off-by: Daniel Engberg <daniel.engberg.lists@pyret.net>
---
 tools/tar/Makefile                                    | 8 ++++----
 tools/tar/patches/100-symlink-force-root-name.patch   | 2 +-
 tools/tar/patches/110-symlink-force-permissions.patch | 2 +-
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/tools/tar/Makefile b/tools/tar/Makefile
index baa431e2db6..2565262d4dd 100644
--- a/tools/tar/Makefile
+++ b/tools/tar/Makefile
@@ -8,13 +8,13 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=tar
 PKG_CPE_ID:=cpe:/a:gnu:tar
-PKG_VERSION:=1.30
+PKG_VERSION:=1.31
 
-PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
 PKG_SOURCE_URL:=@GNU/tar
-PKG_HASH:=87592b86cb037c554375f5868bdd3cc57748aef38d6cb741c81065f0beac63b7
+PKG_HASH:=37f3ef1ceebd8b7e1ebf5b8cc6c65bb8ebf002c7d049032bf456860f25ec2dc1
 
-HOST_BUILD_PARALLEL := 1
+HOST_BUILD_PARALLEL:=1
 
 include $(INCLUDE_DIR)/host-build.mk
 
diff --git a/tools/tar/patches/100-symlink-force-root-name.patch b/tools/tar/patches/100-symlink-force-root-name.patch
index 93f889761b7..8a8ebc0c4cf 100644
--- a/tools/tar/patches/100-symlink-force-root-name.patch
+++ b/tools/tar/patches/100-symlink-force-root-name.patch
@@ -5,7 +5,7 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
 ---
 --- a/src/create.c
 +++ b/src/create.c
-@@ -544,17 +544,8 @@ write_gnu_long_link (struct tar_stat_inf
+@@ -543,17 +543,8 @@ write_gnu_long_link (struct tar_stat_inf
    union block *header;
  
    header = start_private_header ("././@LongLink", size, 0);
diff --git a/tools/tar/patches/110-symlink-force-permissions.patch b/tools/tar/patches/110-symlink-force-permissions.patch
index 6fb799bc0a3..e9a32f7d135 100644
--- a/tools/tar/patches/110-symlink-force-permissions.patch
+++ b/tools/tar/patches/110-symlink-force-permissions.patch
@@ -1,6 +1,6 @@
 --- a/src/create.c
 +++ b/src/create.c
-@@ -1853,6 +1853,7 @@ dump_file0 (struct tar_stat_info *st, ch
+@@ -1852,6 +1852,7 @@ dump_file0 (struct tar_stat_info *st, ch
  #ifdef HAVE_READLINK
    else if (S_ISLNK (st->stat.st_mode))
      {
